### ISSUE TITLE
Sevens rosters from Death Zone

### ISSUE THREAD

**Issue by deanraspa** [BSData/bloodbowl#41]
Are there plans to support sevens rosters as described in the Death Zone book?

**Comment by deanraspa** [comment]
I can fork and attempt to implement if no one is currently working on this.

**Comment by shaner11** [comment]
This was definitely on the todo list, I was just waiting for some spare time. @deanraspa happy for you to get started on it. I have a couple of ideas for implementation and could pass them over in the discord if you'd like.

**Comment by BearSeamen** [comment]
Hey guys. Do you have any plans to move forward on this issue? I can't help with coding, but I can cheer from the sidelines 👯‍♀️

**Comment by Northburns** [comment]
I haven't implemented any BattleScribe data ever (yet), so I am not confident on saying how big and what kind of work this would be. But from what I've gathered, _in a nutshell the differences_ in team roster building are: 

- team size is smaller: 7-11
- Budget is smaller: 600k
- following are more expensive: team re-rolls, assistant coaches, cheerleaders, dedicated fans, apothecary
- there's a separate inducement _smörgåsbord_: not all inducements are available, prices are more expensive, some have stricter max amounts, and there's a new inducement _Desperate Measures_

**Comment by Northburns** [comment]
> Another relevant point: Advancements have higher costs. Basically, your Team values will be off.

Right, totally forgot about advancements! In fact, you can only choose _randomly chosen_ primary/secondary skills, not characteristics. And to make value calculation just a touch trickier, the first skill (whether primary or secondary) has the same value increase as usual, but the rest increase the value more.

**Comment by shaner11** [comment]
Sorry for the long absence. I noticed this pop up again in the discord as well as here. I plan to make a branch to experiment with sevens some time this week and get a feel for how much work it will be. As mentioned, the advancement system is switched up a fair bit and a lot of costs are changed, which makes it harder than just setting a lower roster limit.

I'll try and keep post some updates here. If anybody is willing to help, either by working on the repo or doing some brainstorming let me know.

**Comment by BearSeamen** [comment]
> Sorry for the long absence. I noticed this pop up again in the discord as well as here. I plan to make a branch to experiment with sevens some time this week and get a feel for how much work it will be. As mentioned, the advancement system is switched up a fair bit and a lot of costs are changed, which makes it harder than just setting a lower roster limit.
> 
> I'll try and keep post some updates here. If anybody is willing to help, either by working on the repo or doing some brainstorming let me know.

Thanks for picking this back up. I was desperate enough to look at the repo, but with marginal coding experience, I couldn't make heads or tails of it. 

If there's dull busywork you can hand out, I can probably help. Also, let me know if you still need a clean doc of what's changing. 

As far as brainstorming goes: At least initially, you could skip on positional-based roster verification. Programming wise, it's trivial (every positional increments a counter which can't go past 4), but I have no idea whether you can implement that in battlescribe. Sevens players will know that they can only get 4 positionals.

**Comment by shaner11** [comment]
Quick update from last week. I've laid out the requirements for sevens rosters and started implementing them, and the good news is it won't be much trouble. That being said, it will mean a changes to every catalogue in the system which may result in current rosters breaking a bit. Given that I haven't added freebooter mercenaries yet, I'm going to get that sorted at the same time so I don't have to mess with every single file again down the line. This is also part way through implementation right now, the profiles are there but I need to make each team properly switch between mercenary types.

Along with these, I found a copy of the Dungeon Bowl rules, which have been asked about once or twice. This won't involve major changes to existing files, just adding new teams. It may be part of this update depending on how much time I have to give to it.

Long story short, I should be able to get freebooters and sevens up and running by the end of this week, next week at the latest  if there are interruptions. Dungeon Bowl teams are a stretch goal, but shouldn't be far off if they don't make it in this round of updates.

**Comment by shaner11** [comment]
Unfortunately some things have come up recently that have slowed down my work schedule a bit, so I'm going to release the content that has been ready for the last couple of weeks. The bad news is it doesn't include the Dungeon Bowl teams, as these are obnoxiously time consuming to put together (the College of Life has HOW many player types!?!).

The good news is that LITERALLY EVERYTHING ELSE is good to go. Thank you for your patience on this one. A lot of behind the scenes changes were made to ensure this all worked properly and that regrettably blew the time budget out well beyond my initial estimates (you can blame freebooters, they were a monster to put in).

Apologies to the Dungeon Bowl fans out there, but your time will come soon. I plan to chip away at the remaining 7 teams as I can, so hopefully the wait won't be too long.

With the huge number of changes across the entire game system, it is fairly likely bugs will turn up. Please report these as you find them, and I'll aim to get them in with the Dungeon Bowl Update (or earlier if something is SUPER broken).

**Comment by Zentac** [comment]
@shaner11 any way we can thank you for all the hard work put into this?

**Comment by shaner11** [comment]
Keep on using it so it was worth doing! A big part of the reason I took it up was to support the local player group, so I'm doing it for myself as much as everyone else.

**Comment by Zentac** [comment]
Consider it done!

And thank you ever do much. :)
-------- Oorspronkelijk bericht --------
Op 10 aug. 2022 02:15, schreef crazydude11 :
